src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -67,21 +82,32 @@ public boolean hasClient(Client client) {
     }
 
     /**
-     * Adds a client to the address book.
+     * Replaces the client tag set with a set of matching unique tag objects.
+     */
+    private void replaceClientTagSet(Client client) {
+        Set<Tag> clientLocalTags = client.getTags();
+        tags.addAll(clientLocalTags);
+        client.replaceTags(tags.getTags(clientLocalTags));
+    }
+
```

Comment:
@qwoprocks mentioned that having a one-way transfer of information from Clients to AddressBook is more complicated to implement and so to Synchronise the two set of tags, this method was chosen. 

also may I suggest having a helper function called `synchroniseTagSets()`